lib/osf-components/addon/components/storage-provider-manager/storage-manager/component.ts

```diff
@@ -224,6 +224,11 @@ export default class StorageManager extends Component<Args> {
 
     @action
     selectFile(file: File, event: PointerEvent) {
+        const target = event.target as HTMLElement;
+        // prevent click event from being forwarded to associated <input>
+        if (target?.tagName.toLowerCase() === 'label') {
+            event.preventDefault();
+        }
```

Comment:
Forgive me for being pedantic here, but the "click" event is not exactly forwarded to the associated `<input>`, and `preventDefault` is not really preventing the event being forwarded. What `preventDefault` does is to prevent the browser default actions. It seems that the browser default action for a `click` event on the `<label>` element is to programmatically trigger a `click` event on the input nested in it. And because of event bubbling, the programmatically triggered event will bubble upwards and also trigger the event handler on the `<label>` element. That's why you see the `selectFile` action being called twice: first time because of  real `click` event of the mouse, second time because of the programmatically triggered click event by the browser.
So calling `preventDefault` works here by stopping the second event being triggered programmatically by the browser, not by stopping the first event being forwarded. Event triggered on a parent element will never be forwarded to the child. Hence, you can simply do this:
```suggestion
        event.preventDefault();
```
without having to check for target's `tagName`.

Reply:
I'm explicitly checking the target's tagName because calling `event.preventDefault()` for all target types will prevent the checkbox status toggling for keyboard users hitting the spacebar (in this case, the event target is the `<input>`)

The wording of "forwarded" is a little bit vague/bad, as it could be misconstrued with event-capturing/-bubbling, but yes, this is to prevent the second, automatic "clicking" of the nested checkbox from happening.